apps/test/unit/javalab/JavalabViewTest.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {expect} from '../../util/reconfiguredChai';
+import {shallow} from 'enzyme';
+// We use the UnconnectedJavalabView because shallow().dive() does not work with our version of react-redux
```

Comment:
Keep it to what we know. i.e. we know that calling shallow().dive().instance() on the connected JavalabView fails to give access to the methods owned by JavalabView. You could also say we think it has something to do with the very old version of react-redux we're using. If you do, specify the version so a future dev can check the behavior when the version gets updated. Without a bug report from Enzyme, React, or Redux, however, we don't know this for sure. For example, it's also possible that our manual mocking of the redux store (via apps/src/redux.js) is causing these issues. On the other hand, if you're able to find a bug report describing this behavior, link it here.

A future code maintainer will likely pick up our investigation at some point. Comments like these are intended to give as much information as possible without leading them in the wrong direction.